### DISCUSSION TITLE
Is root's `+layout.svelte` function that `onMount` maybe returns ever called?

### DISCUSSION THREAD

**Discussion by frederikhors** [sveltejs/kit#10669] [Q&A]
Is root's `+layout.svelte` function that [`onMount`](https://svelte.dev/docs/svelte#onmount) maybe returns ever called?

I'm wondering if it makes sense to return something from there since I don't think it ever gets called anyway because I don't see when it can be executed if for example the browser window is closed.

What do you think?

**Comment by frederikhors** [comment]
Nobody answers?

**Reply by alexbjorlig** [reply]
Hi, @frederikhors.

Thanks for taking the time to ask a question. 

I'm unsure what you mean by this part:

>  if it makes sense to return something from there

Can you share some insights into what you are trying to achieve?

**Reply by frederikhors** [reply]
I'm referring to this case:

- file: /src/routes/+layout.svelte

```svelte
<script lang="ts">
	import { onMount } from 'svelte';

	onMount(() => callMe());
</script>

Hello!
```

If `callMe` returns a function would ever be called given that we are in the basic layout of the entire Svelte app?

And if I close the browser window or the PWA window on the device I assume that the eventual function will never be called, right?
